### PR TITLE
Replace sampled tokens with EOD after EOD has been sampled once

### PR DESCRIPTION
**Pull request by vadam5**
Signed-off-by: Virginia Adams <vadams@nvidia.com>

Replace sampled tokens with EOD after EOD has been sampled once

**Collection**: BigNLP

# Changelog 
- Updated 

# Usage
Same as before

# Before your PR is "Ready for review"
**Pre checks**:
- [X] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
- [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/modules/common/text_generation_utils.py [hunk 1]
```diff
@@ -876,6 +879,7 @@ def tab_sample_sequence_batch(
                 prev = torch.clamp(prev, max=tokenizer.vocab_size - 1)
 
                 new_tokens = switch(tokens[:, context_length].view(-1), prev, started)
+                new_tokens = switch(new_tokens, eod_id, is_done)
```
Comment (yidong72) [inline nemo/collections/nlp/modules/common/text_generation_utils.py:882]: For tabular data generation, still preserve the old behavior.

**Review by yidong72 — COMMENTED** [review]
the tabular data generation needs the old behavior. Let's preserve it.

File: nemo/collections/nlp/modules/common/text_generation_utils.py (hunk 1 above)
Comment (vadam5) [reply nemo/collections/nlp/modules/common/text_generation_utils.py:882]: sure thing

**Review by yidong72 — APPROVED** [review]
LGTM

**Pull request merged**